rpc/codec/conn.go

```diff
@@ -9,94 +9,50 @@ import (
 	"encoding/binary"
 	"io"
 	"net"
-	"sync"
-
-	"code.google.com/p/snappy-go/snappy"
 
+	wire "github.com/cockroachdb/cockroach/rpc/codec/wire.pb"
 	"github.com/gogo/protobuf/proto"
 )
 
-type frameBuf struct {
-	size [binary.MaxVarintLen64]byte
+const enableSnappy = true
+
+type decompressFunc func(src []byte, m proto.Message) error
+
+var decompressors = [...]decompressFunc{
```

Comment:
I stand corrected; this does work (although it would be ugly if the constants were large, since it inserts zero values in the array for any numbers from 0 to max() that aren't found).